qiskit/optimization/converters/ising_to_quadratic_program.py

```diff
@@ -94,12 +97,21 @@ def encode(self, qubit_op: WeightedPauliOperator, offset: float = 0.0) -> Quadra
             # The coefficient of the linear term in `QuadraticProgram` is
             # 2 * weight of the pauli
             coef = weight * 2
-            linear[i] = -coef
+            if self._linear:
+                # If the linear option is True, add it into linear_terms
+                linear_terms[i] = -coef
+            else:
+                # Else, add it into quadratic_terms as a diagonal element.
+                quadratic_terms[i, i] = -coef
             # Sub the weight of the linear pauli term from the QUBO matrix
             self._qubo_matrix[i, i] -= weight
             offset += weight
 
-        self._qp.minimize(offset, linear, quadratic)
+        if self._linear:
```

Comment:
This case distinction is not necessary, as "linear_terms" will just be empty if linear == False.

Reply:
Right. I removed it.